Description: optionally inherit PATH from parent process
 The default PATH of cron isn't very useful.  This patch adds an
 option -P to leave PATH alone, so that it's inherited from whatever
 execs cron.  In our case that is daemontools, but could also be
 start-stop-daemon or whatever.  This puts our sysadmins in control
 of cron's (and child processes') PATH, as they are able to set it in the 
 run script.  Without the -P option, crontabs that don't set PATH
 themselves get a value of /usr/bin:/bin, which is hardcoded as
 _PATH_DEFPATH in pathnames.h.
Author: Tom Jones <tom@oxix.org>
Origin: vendor, https://bugs.debian.org/688716
Bug-Debian: https://bugs.debian.org/688716
--- a/cron.8
+++ b/cron.8
@@ -36,6 +36,9 @@ runlevels.
 .B \-f
 Stay in foreground mode, don't daemonize.
 .TP
+.B -P
+Don't set PATH for child processes.  Let it inherit instead.
+.TP
 .B \-l
 Enable LSB compliant names for /etc/cron.d files.  This setting, however, does
 not affect the parsing of files under /etc/cron.hourly, /etc/cron.daily,
--- a/cron.c
+++ b/cron.c
@@ -94,7 +94,9 @@ main(argc, argv)
 	set_cron_cwd();
 
 #if defined(POSIX)
-	setenv("PATH", _PATH_DEFPATH, 1);
+	if (change_child_path) {
+		setenv("PATH", _PATH_DEFPATH, 1);
+	}
 #endif
 
 	/* Get the default locale character set for the mail
@@ -448,9 +450,9 @@ sighup_handler(int x) {
 
 
 #if DEBUGGING
-const char *getoptarg = "flL:nx:";
+const char *getoptarg = "fPlL:nx:";
 #else
-const char *getoptarg = "flL:n";
+const char *getoptarg = "fPlL:n";
 #endif
 
 static void
@@ -461,6 +463,7 @@ parse_args(argc, argv)
 	int	argch;
 
 	stay_foreground = 0;
+	change_child_path = 1;
 	lsbsysinit_mode = 0;
 	log_level = 1;
 	fqdn_in_subject = 0;
@@ -472,6 +475,9 @@ parse_args(argc, argv)
 		case 'f':
 			stay_foreground = 1;
 			break;
+		case 'P':
+			change_child_path = 0;
+			break;
 		case 'l':
 			lsbsysinit_mode = 1;
 			break;
--- a/cron.h
+++ b/cron.h
@@ -306,6 +306,7 @@ time_min clockTime;
 static long GMToff;
 
 int	stay_foreground;
+int	change_child_path;
 int	lsbsysinit_mode;
 int	log_level;
 int	fqdn_in_subject;
--- a/entry.c
+++ b/entry.c
@@ -28,6 +28,7 @@ static char rcsid[] = "$Id: entry.c,v 2.
 
 #include "cron.h"
 
+extern int change_child_path;
 
 typedef	enum ecode {
 	e_none, e_minute, e_hour, e_dom, e_month, e_dow,
@@ -289,7 +290,7 @@ load_entry(file, error_func, pw, envp)
 			goto eof;
 		}
 	}
-	if (!env_get("PATH", e->envp)) {
+	if (!env_get("PATH", e->envp) && change_child_path) {
 		snprintf(envstr, MAX_ENVSTR, "PATH=%s", _PATH_DEFPATH);
 		if ((tenvp = env_set(e->envp, envstr))) {
 			e->envp = tenvp;
